doc/source/tune-usage.rst

```diff
@@ -22,29 +19,48 @@ Start by installing, importing, and initializing Ray.
 
     ray.init()
 
-Tune provides a ``run_experiments`` function that generates and runs the trials as described by the `experiment specification <tune-usage.html#experiment-configuration>`__.
 
-.. autofunction:: ray.tune.run_experiments
-    :noindex:
+Experiment Configuration
+------------------------
 
-This function will report status on the command line until all Trials stop:
+This section will cover the main steps needed to modify your code to run Tune: using the `Training API <tune-usage.html#training-api>`__ and  `executing your Tune experiment <tune-usage.html#specifying-experiments>`__.
 
-::
+You can checkout out our `examples page <tune-examples.html>`__ for more code examples.
 
-    == Status ==
-    Using FIFO scheduling algorithm.
-    Resources used: 4/8 CPUs, 0/0 GPUs
-    Result logdir: ~/ray_results/my_experiment
-     - train_func_0_lr=0.2,momentum=1:  RUNNING [pid=6778], 209 s, 20604 ts, 7.29 acc
-     - train_func_1_lr=0.4,momentum=1:  RUNNING [pid=6780], 208 s, 20522 ts, 53.1 acc
-     - train_func_2_lr=0.6,momentum=1:  TERMINATED [pid=6789], 21 s, 2190 ts, 100 acc
-     - train_func_3_lr=0.2,momentum=2:  RUNNING [pid=6791], 208 s, 41004 ts, 8.37 acc
-     - train_func_4_lr=0.4,momentum=2:  RUNNING [pid=6800], 209 s, 41204 ts, 70.1 acc
-     - train_func_5_lr=0.6,momentum=2:  TERMINATED [pid=6809], 10 s, 2164 ts, 100 acc
+Training API
+~~~~~~~~~~~~
 
+Training can be done with either the **function-based API** or **Trainable API**.
+
+**Python functions** will need to have the following signature:
+
+.. code-block:: python
+
+    def trainable(config, reporter):
+        """
+        Args:
+            config (dict): Parameters provided from the search algorithm
+                or variant generation.
+            reporter (Reporter): Handle to report intermediate metrics to Tune.
+        """
+
+        while True:
+            # ...
+            reporter(**kwargs)
+
+The reporter will allow you to report metrics used for scheduling, search, or early stopping.
+
+Tune will run this function on a separate thread in a Ray actor process. Note that this API is not checkpointable, since the thread will never return control back to its caller. The reporter documentation can be `found here <tune-package-ref.html#ray.tune.function_runner.StatusReporter>`__.
```

Comment:
Hmm should we support checkpointing here eventually like the Tune paper describes?

Reply:
Yeah; it shouldn't be too bad (use the reporter to pass the messages to the main thread).